File: proxy.js
import http from "http";
import fetch from "node-fetch";

http.createServer(async (req, res) => {
  res.setHeader("Access-Control-Allow-Origin", "*");
  res.setHeader("Access-Control-Allow-Headers", "Content-Type");
  res.setHeader("Access-Control-Allow-Methods", "POST, OPTIONS");

  if (req.method === "OPTIONS") {
    res.writeHead(204);
    return res.end();
  }

  let body = "";
  req.on("data", d => body += d);
  req.on("end", async () => {
    const r = await fetch("http://127.0.0.1:16000/json_rpc", {
      method: "POST",
      body,
      headers: { "Content-Type": "application/json" }
    });
    const t = await r.text();
    res.end(t);
  });
}).listen(16001);

File: proxy.py
from http.server import BaseHTTPRequestHandler, HTTPServer
import requests

WALLETD_URL = "http://127.0.0.1:16000/json_rpc"

class Handler(BaseHTTPRequestHandler):
    def do_OPTIONS(self):
        self.send_response(204)
        self.send_header("Access-Control-Allow-Origin", "*")
        self.send_header("Access-Control-Allow-Headers", "Content-Type")
        self.send_header("Access-Control-Allow-Methods", "POST, OPTIONS")
        self.end_headers()

    def do_POST(self):
        length = int(self.headers["Content-Length"])
        body = self.rfile.read(length)

        r = requests.post(
            WALLETD_URL,
            data=body,
            headers={"Content-Type": "application/json"},
            timeout=30
        )

        self.send_response(200)
        self.send_header("Access-Control-Allow-Origin", "*")
        self.send_header("Content-Type", "application/json")
        self.end_headers()
        self.wfile.write(r.content)

print("CORS proxy running on http://127.0.0.1:16001")
HTTPServer(("127.0.0.1", 16001), Handler).serve_forever()

